src/device/generate.py

```diff
@@ -474,7 +474,7 @@ def custom_sort_key(fn):
   out("//   bits 16-19:  ty index\n")
   out("#include <unordered_map>\n")
   out("extern std::unordered_map<uint64_t, int> ncclDevFuncNameToId = {\n")
-  for fn in func_rows:
+  for fn in func_rows[:len(func_rows)//len(all_unroll)]:
```

Comment:
pls add a comment in place here why this division is needed. Also in the `enumerate_func_rows` indicate that unroll has to be at top level.